server/lib/parties/party-service.ts

```diff
@@ -234,8 +239,38 @@ export default class PartyService {
   }
 
   leaveParty(partyId: string, userId: number, clientId: string) {
+    const party = this.parties.get(partyId)
+    if (!party) {
+      throw new PartyServiceError(PartyServiceErrorCode.PartyNotFound, 'Party not found')
+    }
+
+    if (!party.members.has(userId)) {
+      throw new PartyServiceError(PartyServiceErrorCode.UserNotInParty, 'User not in party')
+    }
+
     const clientSockets = this.getClientSockets(userId, clientId)
-    this.removeClientFromParty(clientSockets)
+    this.removeClientFromParty(clientSockets, party)
+  }
+
+  sendChatMessage(partyId: string, userId: number, message: string) {
+    const party = this.parties.get(partyId)
+    if (!party) {
```

Comment:
Same as above, this is revealing far too much information to the user

Reply:
And honestly you should audit all the APIs in here for this, I think they have similar problems.

We're not dealing with trusted clients here, we can't be returning errors like we are. Keep the information limited to what we need for reasonable client behavior.